File: backend/src/models/Meeting.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Meeting = sequelize.define('Meeting', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  lead_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  scheduled_by: {
    type: DataTypes.INTEGER,
    allowNull: false,
    comment: 'User who scheduled the meeting (Level 2)',
  },
  attendees: {
    type: DataTypes.JSON,
    defaultValue: [],
    comment: 'Array of user IDs attending the meeting',
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  description: {
    type: DataTypes.TEXT,
  },
  meeting_date: {
    type: DataTypes.DATE,
    allowNull: false,
  },
  location: {
    type: DataTypes.STRING(255),
  },
  meeting_type: {
    type: DataTypes.ENUM('in_person', 'online', 'phone'),
    defaultValue: 'in_person',
  },
  status: {
    type: DataTypes.ENUM('scheduled', 'completed', 'cancelled', 'postponed'),
    defaultValue: 'scheduled',
  },
  outcome: {
    type: DataTypes.TEXT,
    comment: 'Meeting result/outcome notes',
  },
  next_action: {
    type: DataTypes.TEXT,
    comment: 'Next steps after meeting',
  },
  reminder_sent: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
}, {
  tableName: 'meetings',
  indexes: [
    { fields: ['lead_id'] },
    { fields: ['scheduled_by'] },
    { fields: ['meeting_date'] },
    { fields: ['status'] },
  ],
});

module.exports = Meeting;

File: mobile/lib/models/meeting.dart
class Meeting {
  final int id;
  final String title;
  final String? description;
  final DateTime meetingDate;
  final String? location;
  final String meetingType;
  final String status;
  final String? leadName;
  final String? leadPhone;
  final String? outcome;

  Meeting({
    required this.id,
    required this.title,
    this.description,
    required this.meetingDate,
    this.location,
    required this.meetingType,
    required this.status,
    this.leadName,
    this.leadPhone,
    this.outcome,
  });

  factory Meeting.fromJson(Map<String, dynamic> json) => Meeting(
        id: json['id'],
        title: json['title'],
        description: json['description'],
        meetingDate: DateTime.parse(json['meeting_date']),
        location: json['location'],
        meetingType: json['meeting_type'] ?? 'in_person',
        status: json['status'] ?? 'scheduled',
        leadName: json['lead']?['name'],
        leadPhone: json['lead']?['phone'],
        outcome: json['outcome'],
      );
}

class AppNotification {
  final int id;
  final String title;
  final String? titleAr;
  final String body;
  final String? bodyAr;
  final String type;
  final bool isRead;
  final DateTime createdAt;

  AppNotification({
    required this.id,
    required this.title,
    this.titleAr,
    required this.body,
    this.bodyAr,
    required this.type,
    required this.isRead,
    required this.createdAt,
  });

  factory AppNotification.fromJson(Map<String, dynamic> json) => AppNotification(
        id: json['id'],
        title: json['title'],
        titleAr: json['title_ar'],
        body: json['body'],
        bodyAr: json['body_ar'],
        type: json['type'],
        isRead: json['is_read'] ?? false,
        createdAt: DateTime.parse(json['created_at']),
      );
}

